bindings/resvg-skia/src/lib.rs

```diff
@@ -69,95 +60,89 @@ pub enum FilterQuality {
 }
 
 pub struct Surface {
-    ptr: *mut ffi::skiac_surface,
+    surface: skia_safe::Surface,
     canvas: Canvas,
 }
 
 impl Surface {
     pub fn new_rgba(width: u32, height: u32) -> Option<Surface> {
-        unsafe {
-            Self::from_ptr(ffi::skiac_surface_create_rgba(width as i32, height as i32))
-        }
+        Surface::new_rgba_impl(width, height, skia_safe::AlphaType::Unpremul)
     }
 
     pub fn new_rgba_premultiplied(width: u32, height: u32) -> Option<Surface> {
-        unsafe {
-            Self::from_ptr(ffi::skiac_surface_create_rgba_premultiplied(width as i32, height as i32))
-        }
+        Surface::new_rgba_impl(width, height, skia_safe::AlphaType::Premul)
     }
-
-    pub unsafe fn from_ptr(ptr: *mut ffi::skiac_surface) -> Option<Surface> {
-        if ptr.is_null() {
-            None
-        } else {
-            Some(Surface {
-                ptr,
-                canvas: Canvas(ffi::skiac_surface_get_canvas(ptr))
-            })
-        }
+    
+    pub unsafe fn from_skia_safe_canvas(canvas: &mut skia_safe::Canvas) -> Option<Surface> {
```

Comment:
@therealbnut can we leave this function safe? Otherwise we would have to have an unsafe block in the renderer.

Reply:
Yeah, I think that’s okay as it’s using Skia-safe. I’m happy for you to update it and merge.

Reply:
updated